utils/flywheel_helpers.py

```diff
@@ -379,3 +379,96 @@ def generate_path_to_container(
         # fw_path += append
 
     return fw_path
+
+
+def get_dicom_file_sop(self, fw_object, study_uid, series_uid, sop_uid):
+        """ locates the specific dicom file associated with a given ROI.
+
+        Given an ROI's study UID, series UID, and SOP uid, locate the exact dicom file
+        that the roi corresponds to.
+
+        Args:
+            fw_object (flywheel object): flywheel file or a parent container of a file
+            study_uid (str): the study UID on the ROI
+            series_uid (str): the series UID on the ROI
+            sop_uid (str):  the SOP uid on the ROI
+
+        Returns:
+            dicom_file (str) the name of the dicom file that the ROI is on.
+
+        """
+
+        container_type = fw_object.container_type
+        if container_type == "file":
+            log.debug('working on file')
+            acq = fw_object.parent
+            files = [fw_object]
+            object_name = fw_object.name
+
+        elif container_type == "acquisition":
+            log.debug('working on acquisition')
+            files = fw_object.reload().files
+            object_name = fw_object.label
+
+        elif container_type == "session":
+            # first extract all files from the session.
+            log.debug('working on session')
+            files = []
+            for acq in fw_object.acquisitions():
+                acq = acq.reload()
+                files.extend(acq.files)
+            object_name = fw_object.label
+
+        # Filter by file type, study UID and series UID:
+        files = [f for f in files if f.type == "dicom" and f.info.get("StudyInstanceUID") == study_uid and f.info.get(
+            "SeriesInstanceUID") == series_uid]
+        if len(files) == 0:
+            log.warning(f"No dicom files found in session {object_name} with matching study/series UID "
```

Comment:
nit: Could be acquisition or file, not just session, right?

Reply:
ROI metadata is stored on the session level.  Going into this, ALL we know is that this ROI belongs to a file somewhere in that session.  So given the current workflow we have for the ohif viewer, this will always start at the session level.

Reply:
also we don't need this part of the code.  ooops